### PR TITLE
Enable nncf for multihead classification

### PR DESCRIPTION
**Pull request by sooahleex**
Apply nncf to multihead classification in the evaluation of the model.

### CODE REVIEW

**Comment by sovrasov** [comment]
@sooahleex overall the changes look fine to me, but experiments and tests are required. The tests could be added [here](https://github.com/openvinotoolkit/training_extensions/blob/develop/external/model-preparation-algorithm/tests/test_ote_training.py), but honestly I'm a bit confused by the fact that the NNCF task is still located in deep-object-reid. Shouldn't it be moved to the MPA subfolder?

**Comment by goodsong81** [comment]
> @sooahleex overall the changes look fine to me, but experiments and tests are required. The tests could be added [here](https://github.com/openvinotoolkit/training_extensions/blob/develop/external/model-preparation-algorithm/tests/test_ote_training.py), but honestly I'm a bit confused by the fact that the NNCF task is still located in deep-object-reid. Shouldn't it be moved to the MPA subfolder?

Currently, we have not yet enabled NNCF support for mmcls so that we delegate NNCF optimization for classification tasks to deep-object-reid. (We guarantee checkpoint compatibility for that).
In my opinion, OTE side test is enough for this purpose.
@sooahleex Could you add links to OTE or SC test build that you tested NNCF functionality?
@JihwanEom Could you spot the test for this change? Maybe we have API/CLI tests for this.

File: torchreid/engine/builder.py [hunk 1]
```diff
@@ -4,14 +4,15 @@
 
 from .image import (ImageAMSoftmaxEngine, MultilabelEngine, MultiheadEngine)
 
+NNCF_ENABLED_LOSS = ['softmax', 'am_softmax', 'am_binary']
```
Comment (JihwanEom) [inline torchreid/engine/builder.py]: For multihead classification, loss_name comes as `softmax,asl` , so asl will raise error I think. Could you revise this line to include `asl`?

**Review by JihwanEom — CHANGES_REQUESTED** [review]

**Comment by JihwanEom** [comment]
> @JihwanEom Could you spot the test for this change? Maybe we have API/CLI tests for this.

Yes, I'll test for this PR

File: torchreid/engine/builder.py [hunk 2]
```diff
@@ -4,14 +4,15 @@
...
 
 def build_engine(cfg, datamanager, model, optimizer, scheduler,
                  should_freeze_aux_models=False,
                  nncf_metainfo=None,
                  compression_ctrl=None,
                  initial_lr=None):
     if should_freeze_aux_models or nncf_metainfo:
-        if cfg.loss.name not in ['softmax', 'am_softmax', 'am_binary']:
+        if not any(loss_name in NNCF_ENABLED_LOSS for loss_name in cfg.loss.name.split(',')):
```
Comment (JihwanEom) [inline torchreid/engine/builder.py]: Could you change from `any` to `all`? I think if not enabled loss comes, this if statement should raise error.

**Comment by sooahleex** [comment]
> Currently, we have not yet enabled NNCF support for mmcls so that we delegate NNCF optimization for classification tasks to deep-object-reid. (We guarantee checkpoint compatibility for that). In my opinion, OTE side test is enough for this purpose. @sooahleex Could you add links to OTE or SC test build that you tested NNCF functionality? @JihwanEom Could you spot the test for this change? Maybe we have API/CLI tests for this.

This is the [link](http://validationreports.sclab.intel.com:8004/reports/build_number_report?test_session_build_number=IMPTOps-35498&environment=mclx02) of the SC test build. All the tests are passed well, and when I tested this in UI, it worked out.

**Comment by sovrasov** [comment]
SC build doesn't test NNCF unfortunately, so now we know that this PR at least doesn't break anything and we have to rely on manual testing only.

**Comment by JihwanEom** [comment]
Hi all, NNCF test for multihead classification was conducted on this OTE PR: https://github.com/openvinotoolkit/training_extensions/pull/1192
I could check all the NNCF test cases are passed. If there is some performance issue, we will create another ticket.

**Review by JihwanEom — APPROVED** [review]
LGTM

**Pull request merged**